inst/htmlwidgets/datatables.js

```diff
@@ -166,7 +166,8 @@ HTMLWidgets.widget({
     // in order that the type=number inputs return a number
     $.valHooks.number = {
       get: function(el) {
-        return el.value * 1;
+        var value = el.value;
+        return isNaN(parseFloat(value)) ? "" : value * 1;
```

Comment:
If you use `parseFloat()`, do you still need the `* 1` trick?

```suggestion
        var value = parseFloat(el.value);
        return isNaN(value) ? "" : value;
```

Reply:
Hi,

I've just did some tests. It seems you're right. `parseFloat(string)` seems to be equivalent to `string * 1`.

Reply:
Great. I'll make the change then.